<launch>
  <!-- URDF and TF -->
  <param name="robot_description" command="$(find xacro)/xacro '$(find husky_description)/urdf/husky.urdf.xacro'" />
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

<!--Activate the pose to baselink broadcast script-->
  <node pkg="husky_waypoint_nav" type="optitrack_tf_broadcaster.py" name="optitrack_tf_broadcaster" output="screen" />

  <!-- Joystick + Teleop -->
  <node pkg="joy" type="joy_node" name="joy_node">
    <param name="dev" value="/dev/input/ps4" />
  </node>

  <node pkg="teleop_twist_joy" type="teleop_node" name="teleop_twist_joy">
    <rosparam>
      axis_linear: 1
      axis_angular: 0
      enable_button: 4
      enable_turbo_button: 5
      scale_linear: 0.4
      scale_linear_turbo: 2.0
      scale_angular: 1.4
    </rosparam>
  </node>

  <!-- Move Base -->
  <include file="$(find husky_navigation)/launch/move_base.launch" />

  <!-- Waypoint Collector -->
  <node pkg="husky_waypoint_nav" type="collect_optitrack_waypoints.py" name="collect_optitrack_waypoints">
    <param name="outdoor_waypoint_nav/coordinates_file" value="$(find husky_waypoint_nav)/config/optitrack_waypoints.txt" />
    <param name="outdoor_waypoint_nav/collect_button_num" value="0" />
    <param name="outdoor_waypoint_nav/end_button_num" value="2" />
    <param name="outdoor_waypoint_nav/collect_button_sym" value="cross" />
    <param name="outdoor_waypoint_nav/end_button_sym" value="square" />
  </node>
</launch>
